### PR TITLE
Move test utils in the `src` dir

### PR DESCRIPTION
**Pull request by andreivladbrg**
### Changelog:

refactor: move test utils in src dir 
build: install prb test as npm package
build: remove prb test git module
 ci: install node js dependencies in build and test job
chore: update remappings accordingly

As mentioned in my comment [here](https://github.com/sablier-labs/v2-core/discussions/508#discussioncomment-7340407), importing files from the `~test/` path is not possible in the [NPM package](https://www.npmjs.com/package/@prb/math?activeTab=code) because this folder doesn't exist.

Inevitably, to make this fully compatible with the idea presented in the discussion linked above, I had to install `prb-test` as an NPM package.

### CODE REVIEW

**Comment by andreivladbrg** [comment]
Context: [here](https://github.com/andreivladbrg/prb-math/blob/593cc7fda2259056d7a968ba7ed884f93d050285/src/test/Utils.sol#L11), `StdUtils` is imported, assuming a remappings set for `forge-std/=lib/forge-std/src/ `
exists.

Question: Should `forge-std` also be installed as a node module? 
My initial answer was no, because if there are other projects that need these test utils they are also working on a foundry project (with forge-std remapping considered as the default configuration), otherwise, if one is testing in Hardhat, a contract as a util would not be needed, no? 

Am I missing something and we should  install as NPM package like this `"@forge-std": "github:foundry-rs/forge-std#commit_hash"`?

**Comment by PaulRBerg** [comment]
Thanks @andreivladbrg, just letting you know that it will take me a while until I find the time to pick this PR up.

File: package.json [hunk 1]
```diff
@@ -9,6 +9,9 @@
   "bugs": {
     "url": "https://github.com/PaulRBerg/prb-math/issues"
   },
+  "dependencies": {
```
Comment (PaulRBerg) [inline package.json:12]: Has to be `devDependencies`

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
looks good but could you also please install Forge Std from npm?

**Comment by PaulRBerg** [comment]
> because if there are other projects that need these test utils they are also working on a foundry project (with forge-std remapping considered as the default configuration), otherwise, if one is testing in Hardhat, a contract as a util would not be needed, no?

Foundry should support nested remappings but I agree that it would be worth testing the set-up before merging this PR.

**Comment by andreivladbrg** [comment]
Addressed everything in my latest commits.

I see that the slither job has failed and I am not sure why, is it because we need to add `pnpm install` step?

**Comment by PaulRBerg** [comment]
I've changed my mind about this PR.

Instead of moving the test utils in `src`, we could instead whitelist the `test/utils` directory to be included in the npm package.

We just need to add `test/utils` here:

https://github.com/PaulRBerg/prb-math/blob/abdf0b28909fb2fb149c56cf841ef4d94f916e1e/package.json#L16-L19

**Comment by PaulRBerg** [comment]
Thanks @andreivladbrg for the latest commit, but I would prefer to open a new PR to start with a clean slate!

**Comment by andreivladbrg** [comment]
Clossing in favor of https://github.com/PaulRBerg/prb-math/pull/211

**Pull request closed without merging**